### ISSUE TITLE
Implement Angular pageslide

### ISSUE THREAD

**Issue by sirtastic** [entrity/tipsy-web-public#53]
### Looking into how to best handle the mobile experience.  
It's imperative (especially moving towards an app) that we make this site a great mobile experience. In researching I've found this angular script here which does a great job and looks very simple to implement sidebars that work better than the current fancy-sidebar I've created and can be easily rigged to work with swipe/etc:
https://github.com/dpiccone/ng-pageslide

Coupled with this script for mobile swipe stuff:
http://ryanmullins.github.io/angular-hammer/
**(a PHASE II implementation unless its a quickie)**:

With those we can create a better UX for mobile users.

#### What I need from you

**I just need you to drop in the script and get it setup for 2 sidebars (left / right).**  Let me know what I need to trigger the sliders once you got it setup. 

**I downloaded the files and looked through, it looked simple. it's possible I can implement though I'm not sure where to place the angular stuff, naming conventions, etc. Figured you might want to do that part so I don't messy up any of the code.**

#### Optionally we can use this full JS if you think it'll be better, I can implement if thats the case. It does everything I've listed above but is not angular: https://mango.github.io/slideout/

**Comment by entrity** [comment]
## Pageslide

I've added the code. I've kept it in its own branch, so checkout `feature/sidebars`.

Run to see where the html has changed `git show  95d4ed6c9a27cf862a3ce37156eda14346bfc1f1 -- app/views/layouts/application.html.erb`

1. Remove the previous aside (from `app/views/layouts/application.html.erb`)
1. Add your sidebar contents to `public/application/rightbar.html` and `public/application/leftbar.html`
1. Add triggers for the pageslides:
  - `    <a ng-click="toggle('leftbarOpen')">clickme</a>`
  - `    <a ng-click="toggle('rightbarOpen')">clickme</a>`
1. Add css for the two pageslides & their triggers

Reassigning to you.

## Hammer

I've added the code and included the module in our app, but (just in case you glossed over this part of the doc) to add any swipe/pinch/etc support for non-browser-default actions, you have to add a directive to whatever DOM element you want to target.

----

commit 81d942c5ce7eb10f8197948b8193d482ba87dee5
Date:   Fri Aug 14 19:51:47 2015 -0700

    Add angular-hammer 3rd-party lib. Addresses #53 (github)

commit 95d4ed6c9a27cf862a3ce37156eda14346bfc1f1
Date:   Fri Aug 14 19:01:45 2015 -0700

    Add 3rd-party ng module 'pageslide'. Addresses #53 (github)

**Comment by sirtastic** [comment]
@entrity Checked out `feature/sidebars` , pulled and merged into `feature/design`.  Bower install, rake db:migrate, etc.

Added the stuff from above and at the moment when I click the buttons it does nothing. No error msg, nothing.

I checked in my commit on `feature/design`  --- can you take a look? Also can you be avail and some point tomorrow? Send me a text msg and I'll be at PC to get my environment working.  Few things seem broken.

**Comment by entrity** [comment]
Made a change on `feature/sidebars`

>commit 527f87de2f922e9aaf55944f6ad6b58f35bbf073
>Date:   Sun Aug 16 08:25:19 2015 -0700

>    Fix 'ps-open' attrs in pageslide sidebars. Addresses #53 (github)

Use the following pattern for the togglers:

```
<a ng-click="toggle('leftbarOpen')"><i class="fa fa-bars"></i></a>
```

**Comment by sirtastic** [comment]
@entrity No changes. Sidebars still not opening.

**Comment by entrity** [comment]
@sirtastic : Working for me. Is your problem that the sidebars won't open or won't close? Have you put any content in the sidebars? I don't see any updates to the branch `feature/sidebars`. Would you add a toggler, commit the change, and push it to `feature/sidebars` so I can get a sense of what you're seeing?

I didn't realize that the two patterns were the same. Honestly, I didn't check what the old one was. I just looked at the code correction I made and re-derived what the pattern should be.

Back to you.

**Comment by sirtastic** [comment]
Thought it was in develop branch. My apologies.

**Comment by sirtastic** [comment]
@entrity Can you make each bar toggle a class on "header" called `left-open` (on left toggle click) and `right-open` (on right toggle click).

Seems to be the only way to keep it position:fixed and had to adjust when sidebars are toggled.

**Comment by sirtastic** [comment]
@entrity ^^

**Comment by entrity** [comment]
@sirtastic : still need this? I just want to confirm because the sidebars don't appear to cover the toggles on my environment: 

![image](https://cloud.githubusercontent.com/assets/950824/9421996/b7ec1578-4835-11e5-8fd0-d0174cd7aae2.png)

**Comment by entrity** [comment]
@sirtastic : anyway, if you still need this, please re-open this ticket.

**Comment by sirtastic** [comment]
They work because I've modified the header toolbar to no longer be sticky so I can continue working on the site.  **I still need the class toggle done.**

**Comment by entrity** [comment]
I'm looking for the 'header' element to which you refer. There appears to be none. What element did you have in mind?

**Comment by entrity** [comment]
@sirtastic : I'd have done it myself if I could find the elem you wanted to affect, but ... you can just add this attr to the elem yourself. It'll work unless there's already an `ng-class` on the elem.

```
ng-class="{'left-open':isToggled('leftbarOpen'), 'right-open':isToggled('rightbarOpen')}"
```

**Comment by sirtastic** [comment]
@entrity  `<header>`

**Comment by entrity** [comment]
commit 91000ec08ff90dfb625ce8153c946bfc0bad3fe7
Date:   Tue Aug 25 15:30:58 2015 -0700

    Add right-open and left-open classes to <header>. Closes #53 (github)
